db/migrate/20211222192244_update_combined_advanced_filters.rb

```diff
@@ -0,0 +1,140 @@
+        { name: 'address-foreign', title: 'outside USA', type: 'search' }
+      ]
+    },
+    name: {
+      name: 'name',
+      title: 'Name (Combination)',
+      description: 'Monitoree name',
+      type: 'combination',
+      fields: [
+        { name: 'first-name', title: 'first', type: 'search' },
+        { name: 'last-name', title: 'last', type: 'search' },
+        { name: 'middle-name', title: 'middle', type: 'search' }
+      ]
+    },
+    telephone: {
+      name: 'telephone',
+      title: 'Primary Contact Telephone Number (Text)',
+      description: 'Monitorees with a primary contact telephone number',
+      type: 'search',
+      options: ['Exact Match', 'Contains']
+    }
+  }
+
+  OLD_FILTER_OPTIONS = {
+    'address-foreign': {
+      name: 'address-foreign',
+      title: 'Address (outside USA) (Text)',
+      description: 'Monitoree Address 1, Town/City, Country, Address 2, Postal Code, Address 3 or State/Province (outside USA)',
+      type: 'search'
+    },
+    'address-usa': {
+      name: 'address-usa',
+      title: 'Address (within USA) (Text)',
+      description: 'Monitoree Address 1, Town/City, State, Address 2, Zip, or County within USA',
+      type: 'search'
+    },
+    'first-name': {
+      name: 'first-name',
+      title: 'Name (First) (Text)',
+      description: 'Monitoree first name',
+      type: 'search'
+    },
+    'last-name': {
+      name: 'last-name',
+      title: 'Name (Last) (Text)',
+      description: 'Monitoree last name',
+      type: 'search'
+    },
+    'middle-name': {
+      name: 'middle-name',
+      title: 'Name (Middle) (Text)',
+      description: 'Monitoree middle name',
+      type: 'search'
+    },
+    'telephone-number': {
+      name: 'telephone-number',
+      title: 'Primary Contact Telephone Number (Exact Match) (Text)',
+      description: 'Monitorees with a specified 10 digit primary contact telephone number',
+      type: 'search'
+    },
+    'telephone-number-partial': {
+      name: 'telephone-number-partial',
+      title: 'Primary Contact Telephone Number (Contains) (Text)',
+      description: 'Monitorees with a primary contact telephone number that contains specified digits',
+      type: 'search'
+    }
+  }
+
+  def up
+    relevant_filters.each do |filter|
+      contents = JSON.parse(filter[:contents])
+      contents.each do |content|
+        filter_name = content['filterOption']['name'].to_sym
+        if %i[address-foreign address-usa first-name last-name middle-name].include?(filter_name)
+          content['filterOption'] = NEW_FILTER_OPTIONS[FILTER_MAPPINGS[filter_name].to_sym]
+          content['value'] = [{ name: filter_name, value: content['value']}]
+        elsif %i[telephone-number telephone-number-partial].include?(filter_name)
+          content['filterOption'] = NEW_FILTER_OPTIONS[FILTER_MAPPINGS[filter_name].to_sym]
+          content['additionalFilterOption'] = filter_name == :'telephone-number-partial' ? 'Contains' : 'Exact Match'
```

Comment:
Something seems wrong with the migration, most likely with the migration of the mapping of the JSON. The saved values from the filters are showing up fine after the migration, but the filter fields are not. Did this work for you?

Reply:
hmm I tried it again and it seems to work fine for me. are you referring specifically to the telephone number filters?

Reply:
All of them

Reply:
what are the steps you've been taking to encounter this issue? the way I've been testing it has been:
1) checking out `1.41.0`
2) completely repopulating the database
3) creating these advanced filters
4) checking out this branch and performing the migration
5) verifying the advanced filters

Reply:
Yeah that's what I did as well. I'll run through it again in case I did something wrong in my process

Reply:
I must have made some mistake because it is working for me now